SimTKmath/Geometry/src/ContactGeometry_Sphere.cpp

```diff
@@ -279,6 +279,103 @@ calcGeodesicAnalytical(const Vec3& xP, const Vec3& xQ,
     setGeodesicToArc(e1, e2, radius, angle, geod);
 }
 
+void ContactGeometry::Sphere::Impl::shootGeodesicInDirectionAnalytically(
+    const Vec3& initialPointApprox,
+    const Vec3& initialTangentApprox,
+    Real finalArcLength,
+    int numberOfKnotPoints,
+    const std::function<void(const ContactGeometry::GeodesicKnotPoint&)>&
+        geodesicKnotPointsSink) const
+{
+    SimTK_ERRCHK1_ALWAYS(
+        numberOfKnotPoints >= 2,
+        "ContactGeometry::Sphere::Impl::shootGeodesicInDirectionAnalytically",
```

Comment:
Line breaks on user-facing error messages should be discouraged: it makes grepping the codebase harder when that error turns up in prod